pkg/operator/cleanup/cleanup_controller_test.go

```diff
@@ -0,0 +1,407 @@
+				createTestNamespace(testNamespace),
+				createTestNamespace(testStaleSecretNamespace),
+				testStaleCredentialsRequest(t),
+				testAWSCredsSecret("kube-system", "aws-creds", testRootAWSAccessKeyID, testRootAWSSecretAccessKey),
+				testAWSCredsSecret(testStaleSecretNamespace, testStaleSecretName, testAWSAccessKeyID, testAWSSecretAccessKey),
+				testClusterVersion(),
+				testInfrastructure(testInfraName),
+			},
+			expectDeletion: true,
+			mockRootAWSClient: func(mockCtrl *gomock.Controller) *mockaws.MockClient {
+				mockAWSClient := mockaws.NewMockClient(mockCtrl)
+				return mockAWSClient
+			},
+		},
+		{
+			name: "cleanup stale credentials request in a manual mode",
+			existing: []runtime.Object{
+				testOperatorConfig("Manual"),
+				createTestNamespace(testNamespace),
+				createTestNamespace(testStaleSecretNamespace),
+				testStaleCredentialsRequest(t),
+				testAWSCredsSecret(testStaleSecretNamespace, testStaleSecretName, testAWSAccessKeyID, testAWSSecretAccessKey),
+				testClusterVersion(),
+				testInfrastructure(testInfraName),
+			},
+			expectDeletion: false,
+			mockRootAWSClient: func(mockCtrl *gomock.Controller) *mockaws.MockClient {
+				mockAWSClient := mockaws.NewMockClient(mockCtrl)
+				return mockAWSClient
+			},
+			expectedConditions: []ExpectedCondition{
+				{
+					conditionType: minterv1.StaleCredentials,
+					reason:        "CredentialsNoLongerRequired",
+					status:        corev1.ConditionTrue,
+				},
+			},
+			expectedCOConditions: []ExpectedCOCondition{
+				{
+					conditionType: configv1.OperatorDegraded,
+					status:        corev1.ConditionTrue,
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
+
+			mockRootAWSClient := test.mockRootAWSClient(mockCtrl)
+
+			fakeClient := fake.NewFakeClient(test.existing...)
+			rscr := &ReconcileStaleCredentialsRequest{
+				ReconcileCredentialsRequest: credentialsrequest.ReconcileCredentialsRequest{
+					Client: fakeClient,
+					Actuator: &actuator.AWSActuator{
+						Client: fakeClient,
+						Codec:  codec,
+						Scheme: scheme.Scheme,
+						AWSClientBuilder: func(accessKeyID, secretAccessKey []byte, c client.Client) (minteraws.Client, error) {
+							return mockRootAWSClient, nil
+						},
+					},
+					PlatformType: configv1.AWSPlatformType,
+				},
+			}
+
+			_, err = rscr.Reconcile(reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      testStaleCRName,
+					Namespace: testNamespace,
+				},
+			})
+
+			if err != nil && !test.expectErr {
+				require.NoError(t, err, "Unexpected error: %v", err)
+			}
+			if err == nil && test.expectErr {
+				t.Errorf("Expected error but got none")
+			}
+
+			cr := getCR(fakeClient)
+			if test.expectDeletion {
+				assert.Nil(t, cr, "expected credentials request to be deleted")
+			} else {
+				assert.NotNil(t, cr, "expected credentials request to exist")
```

Comment:
Think you want a `require.NotNil` here b/c if it is nill, the code below will try to access `cr`.

Reply:
right! made the change